src/codegen/solana_accounts.rs

```diff
@@ -167,7 +225,7 @@ pub(super) fn collect_accounts_from_contract(
         recurse_data.next_queue.clear();
     }
 
-    recurse_data.accounts
+    std::mem::swap(&mut recurse_data.functions, &mut ns.functions);
```

Comment:
Can't you just assign rather than swap?

Reply:
Good point. I can try.

Reply:
Yeah this function is hard to grasp. Probable an `iter_mut()` would do as well. If it has to be using swap, can you please add a comment why it is necessary?

Reply:
I added a comment and used the assignment at the end of the function.